drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/main/java/org/drools/workbench/screens/scesim/backend/server/SceSimServiceImpl.java

```diff
@@ -0,0 +1,310 @@
+            invalidateDMOPackageCache.fire(new InvalidateDMOPackageCacheEvent(resource));
+
+            fireMetadataSocialEvents(resource,
+                                     currentMetadata,
+                                     metadata);
+            return resource;
+        } catch (Exception e) {
+            throw ExceptionUtilities.handleException(e);
+        }
+    }
+
+    @Override
+    public void delete(final Path path,
+                       final String comment) {
+        try {
+            deleteService.delete(path,
+                                 comment);
+        } catch (Exception e) {
+            throw ExceptionUtilities.handleException(e);
+        }
+    }
+
+    @Override
+    public Path rename(final Path path,
+                       final String newName,
+                       final String comment) {
+        try {
+            return renameService.rename(path,
+                                        newName,
+                                        comment);
+        } catch (Exception e) {
+            throw ExceptionUtilities.handleException(e);
+        }
+    }
+
+    @Override
+    public Path copy(final Path path,
+                     final String newName,
+                     final String comment) {
+        try {
+            return copyService.copy(path,
+                                    newName,
+                                    comment);
+        } catch (Exception e) {
+            throw ExceptionUtilities.handleException(e);
+        }
+    }
+
+    @Override
+    public Path copy(final Path path,
+                     final String newName,
+                     final Path targetDirectory,
+                     final String comment) {
+        try {
+            return copyService.copy(path,
+                                    newName,
+                                    targetDirectory,
+                                    comment);
+        } catch (Exception e) {
+            throw ExceptionUtilities.handleException(e);
+        }
+    }
+
+    @Override
+    public boolean accepts(final Path path) {
+        return resourceTypeDefinition.accept(path);
+    }
+
+    @Override
+    public List<ValidationMessage> validate(final Path path) {
+        try {
+            final String content = ioService.readAllString(Paths.convert(path));
+            return validate(path,
+                            content);
+        } catch (Exception e) {
+            throw ExceptionUtilities.handleException(e);
+        }
+    }
+
+    @Override
+    public List<ValidationMessage> validate(final Path path,
+                                            final String content) {
+        return doValidation(path,
+                            content);
+    }
+
+    private List<ValidationMessage> doValidation(final Path path,
+                                                 final String content) {
+        try {
+            final KieModule module = moduleService.resolveModule(path);
```

Comment:
Presumably this was copied from the existing Enumeration editor? 

I assume this method code will be replaced in the future.

Reply:
Yep I'm already cleaning the Presenter backend Service pipeline.